compiler/modules/bir/verify.bal

```diff
@@ -70,12 +70,63 @@ public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode c
     foreach BasicBlock b in code.blocks {
         check verifyBasicBlock(cx, b);
     }
+    check verifyGraph(cx, code.blocks, defn.position);
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
+function verifyGraph(VerifyContext vc, BasicBlock[] blocks, Position pos) returns Error? {
+    boolean[] visited = [];
+    visited[blocks.length() - 1] = false;
+    check traveseGraph(vc, blocks, 0, visited, visited, pos);
+    if visited.indexOf(false) != () {
+        return vc.invalidErr("unreachable blocks in function code", pos);
+    }
+}
+
+function traveseGraph(VerifyContext vc, BasicBlock[] blocks, Label label, boolean[] visited, boolean[] currentGraph, Position pos) returns Error? {
+    if currentGraph[label] {
+        return vc.invalidErr("forward edge graph is not acyclic", pos);
+    }
+    if visited[label] {
+        return;
+    }
+    Label? onPanic = blocks[label].onPanic;
+    if onPanic is Label {
+        visited[onPanic] = true;
+    }
+    visited[label] = true;
+    currentGraph[label] = true;
+    Insn insn = check terminator(vc, blocks[label], pos);
+    if insn is BranchInsn {
+        if insn.backward {
```

Comment:
We can't just relay on backwards branch, you have to validate it as well.

Reply:
Ok you are just cheeking it to give  error, that is ok. But still this check needs to happen for each path otherwise the it is not strong enough.